extensions/src/main/java/org/apache/falcon/extensions/store/ExtensionStore.java

```diff
@@ -265,6 +269,28 @@ public String deleteExtensionMetadata(final String extensionName) throws Validat
         }
     }
 
+    public String registerExtensionMetadata(final String extensionName, final String path, final String description)
+        throws URISyntaxException, FalconException, IOException{
+        Configuration conf = new Configuration();
+        URI uri = new URI(path);
+        conf.set("fs.default.name", uri.getScheme() + "://" + uri.getAuthority());
+        FileSystem fileSystem =  HadoopClientFactory.get().createFalconFileSystem(uri);
+        FileStatus[] fileStatusReadMe = fileSystem.listStatus(new Path(uri.getPath() + "/README"));
```

Comment:
Guess it is sufficient to check for the existence of README and check if it is a file.